packages/sdk/src/index.ts

```diff
@@ -42,23 +46,26 @@ export class PipcookClient {
    * @param protocolWithHostname the daemon hostname with protocol, like "http://192.168.1.50"
    * @param port the port
    */
-  constructor(protocolWithHostname = 'http://127.0.0.1', port = 6927) {
+  constructor(protocolWithHostname = 'http://127.0.0.1', port = 6927, opts?: InitOption) {
+    this.onError = opts?.onError;
     this.endpoint = `${protocolWithHostname}:${port}/api`;
-    this.pipeline = new Pipeline(this.endpoint);
-    this.job = new Job(this.endpoint);
-    this.plugin = new Plugin(this.endpoint);
+    this.pipeline = new Pipeline(this.endpoint, opts);
+    this.job = new Job(this.endpoint, opts);
+    this.plugin = new Plugin(this.endpoint, opts);
```

Comment:
How about sharing the same `BaseApi` in instances of a client?

Reply:
If the onError function is static, these codes could be simplified.

Reply:
This is unrelated to the function `onerror`, the only difference is the route path, they could be maintained at `Pipeline`, `Job`, and `Plugin`.